rust/common/error/macros.rs

```diff
@@ -109,3 +124,33 @@ macro_rules! error_messages {
         }
     };
 }
+
+#[cfg(test)]
+mod tests {
+    error_messages! { TestError
+        code: "TST", type: "Test Error",
+        BasicError() =
+            1: "This is a basic error.",
+        ErrorWithAttributes(i32, String) =
+            2: "This is an error with i32 {} and string '{}'.",
+        MultiLine() =
+            3: "This is an error,\nthat spans,\nmultiple lines."
+    }
+
+    #[test]
+    pub fn debug_includes_display() {
+        let errors = [
+            TestError::BasicError(),
+            TestError::ErrorWithAttributes(1, "error message".to_string()),
+            TestError::MultiLine(),
+        ];
+
+        for error in errors {
+            let display = format!("{error}");
+            let compact_debug = format!("{error:?}");
+            let expanded_debug = format!("{error:#?}");
+            assert!(compact_debug.contains(&display.replace('\n', &"\\n")));
+            assert!(expanded_debug.contains(&display.replace('\n', &"\\n")));
```

Comment:
How is this not failing? `display` includes e.g. `[TST1]` at the start which the debug format omits, as far as I can tell.

Reply:
The `message` field in `std::Debug` has the entire error message produced by `std::Display`.